### ISSUE TITLE
[clangd] [modules] clangd uses stale PCM after prerequisite module construction fails

### ISSUE THREAD

**Issue by berkaysahiin** [llvm/llvm-project#213499]
When processing a consumer with `--experimental-modules-support`, clangd attempts to build its own PCM for each imported module. 

If the module interface listed in the compilation database no longer exists, this step fails. However, clangd then continues processing the consumer using the existing PCM specified by the consumer’s original -fmodule-file argument.

This fallback is not reported to the user, even though the existing PCM may no longer represent the current project state.

1. Build a named module and consumer.
2. Delete the module interface, leaving the CDB and PCM unchanged.
3. Open the consumer in clangd.

Logs:
```
Scanning modules dependencies for M.cppm failed: no such file
Failed to build module M
```
However, it retains -fmodule-file=M=.../M.pcm, loads that PCM, and publishes no diagnostic.

Expected: Report the failed module construction and potentially stale PCM fallback.
Tested with clangd 22.1.8 and --experimental-modules-support.

**Comment by ChuanqiXu9** [comment]
This seems to be a bug but I don't know why. Would you like to work on your local reproducer and send a patch?

**Comment by HighCommander4** [comment]
This behaviour has been noticed before (see e.g. [this comment](https://github.com/llvm/llvm-project/issues/126350#issuecomment-4140545068)).

I agree that it does not seem ideal; it may be better to introduce a restriction that with `--experimental-modules-support`, clangd only looks at PCMs which it has built itself.

**Comment by berkaysahiin** [comment]
@ChuanqiXu9  Yes, I can work on a patch.
